get_element_cpo/P2769.bs

```diff
@@ -429,150 +438,128 @@ wider range of types rather than just the listed standard types.
+        };
+        // using __type = std::remove_cvref_t<_TupleLike>'
+        // if constexpr (__type is std::tuple or std::pair or std::array or
+        //                 std::ranges::subrange)
+        // {
+        //     std::get<_Ip>(std::forward<_TupleLike>(__tuple_like));
+        // }
+        if constexpr (__get_member_well_formed)
+        {
+            return std::forward<_TupleLike>(__tuple_like).template get<_Ip>();
+        }
         return get<_Ip>(std::forward<_TupleLike>(__tuple_like));
     }
 };
 } // namespace __detail
-
-inline namespace __get_fn_namespace
-{
-template <std::size_t _Ip, typename... Args>
-inline constexpr __detail::__get_fn<_Ip, Args...> get;
-} // inline namespace __get_fn_namespace
-} // namespace ranges
 } // namespace std
 ```
 
-With such an implementation, all important cases from our perspective continue working:
-* `std::ranges::get<0>(sub_r)`
-* `std::get<0>(sub_r)`
-* `get<0>(sub_r)`
-* `std::ranges::get<0, some_arg>(sub_r)` (can also be used without namespace)
-
-where `sub_r` is `std::ranges::subrange` object.
-
-The API breaking change appears when `get` has all explicit template arguments for `subrange`,
-i.e., `std::ranges::get<Iarg, Sarg, Karg>(std::ranges::subrange<Iarg, Sarg, Karg>{})`. The problem is
-with the last `Karg` argument, which is unrelated to `tuple_size_v` and `tuple_element_t` of the subrange.
-Even if we say that the proposed backward compatible CPO with `Args...` does not constraint `sizeof...(Args)`
-and ignores the tail outside `tuple_size_v<subrange>`, it doesn't help
-for the mentioned use case because `K` of `std::ranges::subrange` is a non-type template parameter.
-Anyway, this scenario doesn't look common because explicit template parameters are used relatively rarely
-and furthermore, `K` has the default argument that is substituted based on a sentinel.
-
-Definitely such a change would break the ABI for `std::ranges::get` because the fully qualified name of
-this function would change from what is in C++ standard library implementations today.
-But we think that that ABI break would have low impact because `std::ranges::get` is likely
-inlined and so doesn't create a visible symbol in translation units.
-We could imagine other tricky scenario where the API might be broken when `std::ranges::get` is used for something else but call. For example:
-`&std::ranges::get<0>`, `decltype(std::ranges::get<0>)`, etc. However, these scenarios don't look common.
-
-Since the `std::ranges::subrange` API is relatively new, perhaps only a small amount
-of users would be affected but it can not be predicted accurately.
+However, such potential change is very unlikely. In particular, there is no user-defined type involved. If such code was deployed, it's likely to be already broken. Indeed, while `std::get` is not a customisation point, it is often used unqualified (and therefore found by ADL), such that the code above can already be ambiguous.
```

Comment:
> In particular, there is no user-defined type involved.

I do not understand this note.

> If such code was deployed, it's likely to be already broken. ... the code above can already be ambiguous.

I guess it refers to the user code with `my::type`; however, the code right above is a possible implementation modification. It is therefore confusing what "such code / the code above" refers to.